streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -109,67 +108,63 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
                                  this);
     }
 
-    @SuppressWarnings("unchecked")
-    public <K, V, S extends StateStore> KTable<K, V> table(final String topic,
-                                                           final ConsumedInternal<K, V> consumed,
-                                                           final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
-
-        final StoreBuilder<S> storeBuilder = (StoreBuilder<S>) new KeyValueStoreMaterializer<>(materialized).materialize();
-
-        final String source = newProcessorName(KStreamImpl.SOURCE_NAME);
-        final String name = newProcessorName(KTableImpl.SOURCE_NAME);
-        final ProcessorSupplier<K, V> processorSupplier = new KTableSource<>(storeBuilder.name());
-        final ProcessorParameters processorParameters = new ProcessorParameters<>(processorSupplier, name);
-
-        final TableSourceNode.TableSourceNodeBuilder<K, V, S> tableSourceNodeBuilder = TableSourceNode.tableSourceNodeBuilder();
-
-        final TableSourceNode<K, V, S> tableSourceNode = tableSourceNodeBuilder.withNodeName(name)
-                                                                               .withSourceName(source)
-                                                                               .withStoreBuilder(storeBuilder)
-                                                                               .withConsumedInternal(consumed)
-                                                                               .withProcessorParameters(processorParameters)
-                                                                               .withTopic(topic)
-                                                                               .build();
+    public <K, V> KTable<K, V> table(final String topic,
```

Comment:
Removed `S` template since we will always need a key-value store for this API (the MaterializedInternal indicates that already).